bug(pulse): _ci_repair_launch_worker WORKER_GITHUB_LOGIN empty — AIDEVOPS_PULSE_RUNNER_LOGIN has no auto-detect fallback (regression in #30105 fix)

## Description

`_ci_repair_launch_worker` in `pulse-merge-feedback.sh` passes `WORKER_GITHUB_LOGIN="${AIDEVOPS_PULSE_RUNNER_LOGIN:-}"` to the CI repair worker env block. PR #30111 (Resolves #30105) added this line as the fix, but `AIDEVOPS_PULSE_RUNNER_LOGIN` has no default and no auto-detection fallback. If the user has not manually exported this variable, `WORKER_GITHUB_LOGIN` is empty, the ownership fence in `headless-runtime-worker-prepare.sh` rejects the worker with `runner=missing`, and every linked-issue CI repair silently fails.

By contrast, the regular dispatch path (`pulse-dispatch-engine.sh:538`) auto-detects the login via `gh api user --jq '.login'` and passes it to all regular workers. CI repair workers have no equivalent.

## Expected Behavior

`_ci_repair_launch_worker` should resolve the runner login the same way `pulse-dispatch-engine.sh` does — via `gh api user --jq '.login'` — falling back to `AIDEVOPS_PULSE_RUNNER_LOGIN` if set. No manual configuration should be required.

## Reproducer

**Symptom command**:

```
cat /tmp/worker-ci-repair-5e81b355e91d42d3a32f778966fd2ba21e53fd9f72bf28f26b45b24289917c33-3918-d6541dbae8dc.log
```

**Actual output** (from `/tmp/worker-ci-repair-*-3918-*.log`):

```
[INFO] [lifecycle] post_canary session=ci-repair-... model=openai/gpt-5.6-terra
[INFO] [lifecycle] pre_worker_prepare session=ci-repair-...
[ownership-fence] PR_REPAIR_TARGET_VALID: pr=#3918 repo=<repo> head_sha=d6541dbae8dc... head_ref=feature/auto-20260814-144329-gh3890
[ERROR] [ownership-fence] incomplete worker ownership contract issue=3890 repo=<repo> runner=missing
[WARNING] [exit-trap] runtime invocation never started after worker preparation; reason=worker_ownership_lost
```

**Pulse log** (confirms dispatch then immediate failure):

```
[pulse-wrapper] _dispatch_ci_fix_worker: dispatched in-place CI repair for PR #3918 head d6541dbae8dc in <repo>
```

**Expected output**: worker should pass ownership fence and proceed to runtime invocation.

**Causal status**: confirmed

**Owning-path proof**:

- **Production entry point**: `pulse-merge-feedback.sh:_dispatch_ci_fix_worker` → `_ci_repair_launch_worker` called when a non-required CI check is a terminal failure on a worker PR
- **Call chain**: `pulse-wrapper.sh:_dispatch_ci_fix_worker` → `pulse-merge-feedback.sh:_ci_repair_launch_worker:1149` → `env ... WORKER_GITHUB_LOGIN="${AIDEVOPS_PULSE_RUNNER_LOGIN:-}" headless-runtime-helper.sh run` → `headless-runtime-worker-prepare.sh:_hrw_verify_dispatch_ownership:150-153` → `runner_login` empty → `return 1` → `worker_ownership_lost`
- **Integrated verification**: `test-pulse-merge-ci-repair-routing.sh:60` already exports `AIDEVOPS_PULSE_RUNNER_LOGIN="expected-runner"` to make tests pass — proving the variable is required but not documented or auto-set in production

**Causal code**:

```
pulse-merge-feedback.sh:1176
WORKER_GITHUB_LOGIN="${AIDEVOPS_PULSE_RUNNER_LOGIN:-}" \
```

`pulse-dispatch-engine.sh:538` (working path for comparison):

```
self_login=$(gh api user --jq '.login' 2>/dev/null || echo "")
```

**Call-site sweep**:

```
<local-path>
<local-path>  (uses same var, also no auto-detect)
```

## Steps to Reproduce

1. Have a worker PR with a non-required CI failure (e.g. Regression Evidence FAIL)
2. Ensure `AIDEVOPS_PULSE_RUNNER_LOGIN` is not set in the environment
3. Wait for pulse to call `_dispatch_ci_fix_worker` for that PR
4. Observe `worker_ownership_lost` in the ci-repair log under `/tmp/worker-ci-repair-*.log`

## Workarounds Applied

- Manually added `export AIDEVOPS_PULSE_RUNNER_LOGIN=katarzynacc` to `~/.config/aidevops/tenants/default/credentials.sh` — this works but requires knowing the correct GitHub login and is not documented anywhere as a required step

## Proposed Fix

In `_ci_repair_launch_worker`, resolve the runner login with the same pattern used by `pulse-dispatch-engine.sh`:

```bash
local _ci_repair_runner_login="${AIDEVOPS_PULSE_RUNNER_LOGIN:-}"
if [[ -z "$_ci_repair_runner_login" ]]; then
    _ci_repair_runner_login=$(gh api user --jq '.login' 2>/dev/null || true)
fi
```

Then use `"$_ci_repair_runner_login"` in the env block. Same fix should be evaluated for `pulse-campaign-shadow.sh:79`.

## Environment

- **aidevops version**: 3.32.264
- **Latest version**: 3.32.264
- **AI Assistant**: OpenCode
- **OS**: Ubuntu 24.04.4 LTS
- **Shell**: bash 5.2.21(1)-release
- **gh CLI**: gh version 2.89.0 (2026-03-26)

## Additional Context

PR #30111 (merged 2026-08-12, Resolves #30105) added the `WORKER_GITHUB_LOGIN` line to the env block — that was a partial fix. The remaining gap is that `AIDEVOPS_PULSE_RUNNER_LOGIN` has no default and no auto-detection. The test suite papers over this by exporting the variable explicitly (line 60 of the test file), which masked the production gap.

See-also: #30105 (parent issue, partially addressed)


<!-- aidevops:origin:interactive -->
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.32.264 plugin for [OpenCode](https://opencode.ai) v1.18.18 with claude-sonnet-4-6 spent 1h 19m and 55963 tokens on this with the user in an interactive session.

--- a/.agents/scripts/pulse-merge-feedback.sh
+++ b/.agents/scripts/pulse-merge-feedback.sh
@@ -1144,6 +1144,32 @@ _ci_repair_session_identity() {
 }
 
 #######################################
+# Resolve CI repair identity from the active gh credential context.
+# Caller-provided environment identity is never accepted as authority.
+#######################################
+_ci_repair_resolve_runner_login() {
+	local runner_login=""
+
+	# #aidevops:trust-boundary — bind worker ownership to authenticated GitHub
+	# identity rather than a mutable environment variable.
+	runner_login=$(gh api user --jq '.login // ""' 2>/dev/null || true)
+	if [[ "$runner_login" =~ ^[A-Za-z0-9]([A-Za-z0-9-]{0,37}[A-Za-z0-9])?$ ]]; then
+		printf '%s\n' "$runner_login"
+		return 0
+	fi
+
+	runner_login=$(gh api graphql \
+		-f 'query=query { viewer { login } }' \
+		--jq '.data.viewer.login // ""' 2>/dev/null || true)
+	if [[ "$runner_login" =~ ^[A-Za-z0-9]([A-Za-z0-9-]{0,37}[A-Za-z0-9])?$ ]]; then
+		printf '%s\n' "$runner_login"
+		return 0
+	fi
+
+	return 1
+}
+
+#######################################
 # Launch through the runtime's native detach path and publish its process identity.
 #######################################
 _ci_repair_launch_worker() {
@@ -1164,16 +1190,21 @@ _ci_repair_launch_worker() {
 	local process_start=""
 	local session_identity=""
 	local dispatched_status=""
+	local runner_login=""
 	local wait_count=0
 	local wait_max="${AIDEVOPS_CI_REPAIR_SESSION_LOCK_WAIT_STEPS:-200}"
 	local process_pattern="${WORKER_PROCESS_PATTERN:-opencode|claude|Claude}|headless-runtime-helper"
 
 	[[ "$wait_max" =~ ^[0-9]+$ ]] || wait_max=200
+	runner_login=$(_ci_repair_resolve_runner_login) || {
+		printf '%s\n' "[pulse-wrapper] _ci_repair_launch_worker: authenticated GitHub runner identity unavailable for ${repo_slug} PR #${pr_number}; launch blocked" >>"$LOGFILE"
+		return 1
+	}
 	launch_output=$(env \
 		HEADLESS=1 WORKER_ISSUE_NUMBER="$linked_issue" WORKER_REPO_SLUG="$repo_slug" \
 		WORKER_WORKTREE_PATH="$worktree_path" GITHUB_REPOSITORY="$repo_slug" \
 		WORKER_NO_EXIT_PUSH=1 WORKER_PROCESS_PATTERN="$process_pattern" AIDEVOPS_ALLOW_WORKER_WORKTREE_OWNER_TRANSFER=1 \
-		WORKER_GITHUB_LOGIN="${AIDEVOPS_PULSE_RUNNER_LOGIN:-}" \
+		WORKER_GITHUB_LOGIN="$runner_login" \
 		AIDEVOPS_PR_REPAIR_NUMBER="$pr_number" AIDEVOPS_PR_REPAIR_HEAD_SHA="$pr_head_sha" \
 		AIDEVOPS_PR_REPAIR_HEAD_REF="$pr_head_ref" AIDEVOPS_PR_REPAIR_FINGERPRINT="$failure_fingerprint" \
 		AIDEVOPS_PR_REPAIR_OWNERSHIP_MODE="linked-issue" \

--- a/.agents/scripts/tests/test-pulse-merge-ci-repair-routing.sh
+++ b/.agents/scripts/tests/test-pulse-merge-ci-repair-routing.sh
@@ -53,11 +53,15 @@ setup_test_env() {
 	unset TEST_INITIAL_PR_HEAD_SHA
 	unset TEST_INITIAL_PR_HEAD_EMPTY
 	unset TEST_WORKTREE_ADD_FAIL
+	unset TEST_GH_REST_FAIL
+	unset TEST_GH_REST_LOGIN
+	unset TEST_GH_GRAPHQL_FAIL
+	unset TEST_GH_GRAPHQL_LOGIN
+	unset AIDEVOPS_PULSE_RUNNER_LOGIN
 	export AIDEVOPS_CI_REPAIR_STATE_DIR="${TEST_ROOT}/repair-state"
 	export AIDEVOPS_CI_REPAIR_WORKTREE_BASE_DIR="${TEST_ROOT}/worktrees"
 	export AIDEVOPS_HEADLESS_RUNTIME_DIR="${TEST_ROOT}/headless-runtime"
 	export AIDEVOPS_CI_REPAIR_SESSION_LOCK_WAIT_STEPS="0"
-	export AIDEVOPS_PULSE_RUNNER_LOGIN="expected-runner"
 	mkdir -p "${TEST_ROOT}/repo"
 	TEST_PR_HEAD_SHA="abcdef0123456789abcdef0123456789abcdef01"
 	export TEST_PR_HEAD_SHA
@@ -204,9 +208,28 @@ if [[ "${1:-} ${2:-}" == "run view" ]]; then
 	esac
 	exit 0
 fi
+
 GHEOF
+	_append_gh_auth_mock_routes
 	_append_gh_mock_routes
 	chmod +x "${TEST_ROOT}/bin/gh"
+	return 0
+}
+
+_append_gh_auth_mock_routes() {
+	cat >>"${TEST_ROOT}/bin/gh" <<'GHEOF'
+	if [[ "${1:-} ${2:-}" == "api user" ]]; then
+		[[ "${TEST_GH_REST_FAIL:-0}" == "1" ]] && exit 1
+		printf '%s\n' "${TEST_GH_REST_LOGIN:-expected-runner}"
+		exit 0
+	fi
+
+	if [[ "${1:-} ${2:-}" == "api graphql" ]]; then
+		[[ "${TEST_GH_GRAPHQL_FAIL:-0}" == "1" ]] && exit 1
+		printf '%s\n' "${TEST_GH_GRAPHQL_LOGIN:-expected-runner}"
+		exit 0
+	fi
+GHEOF
 	return 0
 }
 
@@ -443,6 +466,7 @@ define_feedback_helpers() {
 		_ci_repair_claim_lease
 		_ci_repair_create_worktree
 		_ci_repair_session_identity
+		_ci_repair_resolve_runner_login
 		_ci_repair_launch_worker
 		_ci_repair_write_prompt
 		_ci_repair_legacy_lease_is_active
@@ -755,6 +779,57 @@ test_changes_requested_empty_labels_refresh_current_metadata() {
 		print_result "refreshed empty PR labels preserve review routing" 1 "route_calls=${ROUTE_CALLS}, route_args=${ROUTE_ARGS}"
 	else
 		print_result "empty PR labels refresh current metadata before review routing" 0
+	fi
+	teardown_test_env
+	return 0
+}
+
+test_ci_repair_resolves_authenticated_runner_identity() {
+	setup_test_env
+	define_feedback_helpers || { print_result "defines authenticated CI repair identity helpers" 1 "could not extract feedback helpers"; teardown_test_env; return 0; }
+
+	local resolved_login=""
+	local rc=0
+	resolved_login=$(AIDEVOPS_PULSE_RUNNER_LOGIN="forged-runner" _ci_repair_resolve_runner_login) || rc=$?
+	if [[ "$rc" -eq 0 && "$resolved_login" == "expected-runner" ]]; then
+		print_result "CI repair identity ignores caller environment and uses authenticated REST viewer" 0
+	else
+		print_result "CI repair identity ignores caller environment and uses authenticated REST viewer" 1 "login=${resolved_login:-missing}, rc=${rc}"
+	fi
+
+	rc=0
+	resolved_login=$(TEST_GH_REST_FAIL=1 TEST_GH_GRAPHQL_LOGIN="graphql-runner" _ci_repair_resolve_runner_login) || rc=$?
+	if [[ "$rc" -eq 0 && "$resolved_login" == "graphql-runner" ]]; then
+		print_result "CI repair identity uses authenticated GraphQL viewer fallback" 0
+	else
+		print_result "CI repair identity uses authenticated GraphQL viewer fallback" 1 "login=${resolved_login:-missing}, rc=${rc}"
+	fi
+
+	mkdir -p "${TEST_ROOT}/lease-success"
+	: >"$GH_LOG"
+	rc=0
+	AIDEVOPS_PULSE_RUNNER_LOGIN="forged-runner" _ci_repair_launch_worker \
+		"${TEST_ROOT}/lease-success" "$AIDEVOPS_HEADLESS_RUNTIME_HELPER" "owner/repo" "100" "42" \
+		"$TEST_PR_HEAD_SHA" "feature/repair" "fingerprint" "${TEST_ROOT}/worktrees/repair" \
+		"1" "ci-repair-auth-success" "${TEST_ROOT}/prompt.md" || rc=$?
+	if [[ "$rc" -eq 0 ]] && grep -qF '|expected-runner|run --role worker' "$GH_LOG"; then
+		print_result "authenticated CI repair identity reaches worker launch" 0
+	else
+		print_result "authenticated CI repair identity reaches worker launch" 1 "rc=${rc}, log=$(<"$GH_LOG")"
+	fi
+
+	: >"$GH_LOG"
+	rc=0
+	TEST_GH_REST_FAIL=1 TEST_GH_GRAPHQL_FAIL=1 _ci_repair_launch_worker \
+		"${TEST_ROOT}/lease" "$AIDEVOPS_HEADLESS_RUNTIME_HELPER" "owner/repo" "100" "42" \
+		"$TEST_PR_HEAD_SHA" "feature/repair" "fingerprint" "${TEST_ROOT}/worktrees/repair" \
+		"1" "ci-repair-auth-test" "${TEST_ROOT}/prompt.md" || rc=$?
+	if [[ "$rc" -eq 0 ]]; then
+		print_result "CI repair launch fails closed without authenticated identity" 1 "launch unexpectedly succeeded"
+	elif grep -qF 'run --role worker' "$GH_LOG"; then
+		print_result "CI repair launch fails closed without authenticated identity" 1 "worker helper was invoked"
+	else
+		print_result "CI repair launch fails closed without authenticated identity" 0
 	fi
 	teardown_test_env
 	return 0
@@ -1225,6 +1300,7 @@ main() {
 	test_rest_missing_review_decision_refreshes_before_ci_route
 	test_coderabbit_nits_ok_dismissed_once_before_late_gate
 	test_changes_requested_empty_labels_refresh_current_metadata
+	test_ci_repair_resolves_authenticated_runner_identity
 	test_ci_repair_dedupes_identical_evidence_for_same_head
 	test_ci_repair_dedupes_changed_evidence_for_same_head
 	test_ci_repair_respects_live_legacy_lease

